cryostatio/cryostat3
# fix(reports): report generation does not cause recursive update

# Welcome to Cryostat3! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #367

## Description of the change:
See the linked bug. There is a Quarkus change that causes the `default` method implementations in the `ReportsService` to break the dependency injection system such that the memory caching layer decorator receives an instance of itself for injection, rather than the storage cache layer as a delegate. This causes it to recursively call itself, causing the original bug. Lifting the default method implementations out of the interface and into the concrete implementations fixes the injection bug.

## How to manually test:
1. Run smoketest
2. Try to generate a report. It should succeed
